src/components/EmailForm/PopupContent.tsx

```diff
@@ -0,0 +1,41 @@
+import { ERROR_MESSAGE, SUCCESS_MESSAGE } from '../../utils/constants';
+import { Button } from '../Button';
+
+type PopupContentProps = {
+  status: number | undefined;
+  onClose: () => void;
+};
+
+export const PopupContent: React.FC<PopupContentProps> = ({
+  status,
+  onClose,
+}) => {
+  let title;
+
+  switch (status) {
+    case 401:
+      title = ERROR_MESSAGE.AUTHENTICATION_REQUIRED;
+      break;
+    case 409:
+      title = ERROR_MESSAGE.REQUEST_ALREADY_SENT;
+      break;
+    case 200:
+      title = SUCCESS_MESSAGE.REQUEST_SENT;
+      break;
+    default:
+      title = ERROR_MESSAGE.GENERIC_ERROR;
+  }
+
+  return (
+    <>
+      <h2
+        className={`mt-5 ${status === 200 ? 'text-primary' : 'text-secondary'}`}
+      >
+        {title}
```

Comment:
```suggestion

const getTitle = (status: number | undefined) => {
  switch (status) {
    case 401:
     return ERROR_MESSAGE.AUTHENTICATION_REQUIRED;
    case 409:
      return ERROR_MESSAGE.REQUEST_ALREADY_SENT;
    case 200:
      return SUCCESS_MESSAGE.REQUEST_SENT;
    default:
      return ERROR_MESSAGE.GENERIC_ERROR;
  }

}

type PopupContentProps = {
  status: number | undefined;
  onClose: () => void;
};

export const PopupContent: React.FC<PopupContentProps> = ({
  status,
  onClose,
}) => {

  return (
    <>
      <h2
        className={`mt-5 ${status === 200 ? 'text-primary' : 'text-secondary'}`}
      >
        {getTitle(status)}
```

and now you have it without local mutation (local is fine). I find this easier to read, cause you came to read what popupcontent renders, if you can how TITLE is rendered, you would continue that way.

Reply:
Done. ✅